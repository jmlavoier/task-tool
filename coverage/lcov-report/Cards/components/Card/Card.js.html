<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Cards/components/Card/Card.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">Cards/components/Card</a> Card.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">28.57% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/12</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { DragSource } from 'react-dnd';
import classNames from 'classnames';
&nbsp;
import CardForm from 'modules/Cards/components/CardForm';
import style from './Card.sass';
&nbsp;
const cardSource = {
  beginDrag(props) {
<span class="cstat-no" title="statement not covered" >    return {</span>
      id: props.id,
    };
  },
  endDrag(props, monitor) {
<span class="cstat-no" title="statement not covered" >    if (!monitor.getDropResult()) {</span>
<span class="cstat-no" title="statement not covered" >      return;</span>
    }
    const { boardId } = <span class="cstat-no" title="statement not covered" >monitor.getDropResult();</span>
    const { id, changeBoard } = <span class="cstat-no" title="statement not covered" >props;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    changeBoard(id, boardId);</span>
  },
};
&nbsp;
function <span class="fstat-no" title="function not covered" >collect(</span>connect, monitor) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging(),
  };
}
&nbsp;
const getClassNames = <span class="fstat-no" title="function not covered" >is</span>Dragging =&gt; <span class="cstat-no" title="statement not covered" >classNames({</span>
  [style.card]: true,
  [style['is-dragging']]: isDragging,
});
&nbsp;
const Card = <span class="fstat-no" title="function not covered" >({</span> id, description, editMode, onSaveCard, onEditCard, isDragging, connectDragSource }) =&gt; <span class="cstat-no" title="statement not covered" >connectDragSource(</span>
  &lt;div role="presentation" className={getClassNames(isDragging)} onClick={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >onEditCard(id)}</span> &gt;
    {editMode
        ? &lt;CardForm id={id} description={description} onSaveCard={onSaveCard} /&gt;
        : description}
  &lt;/div&gt;,
);
&nbsp;
Card.propTypes = {
  description: PropTypes.string.isRequired,
  isDragging: PropTypes.bool.isRequired,
  connectDragSource: PropTypes.func.isRequired,
  changeBoard: PropTypes.func.isRequired,
};
&nbsp;
export default DragSource('CARD', cardSource, collect)(Card);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Nov 04 2017 19:30:25 GMT-0200 (-02)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
