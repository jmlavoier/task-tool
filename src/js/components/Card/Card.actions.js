export const CREATE_CARD = 'CREATE_CARD';
export const UPDATE_CARD = 'UPDATE_CARD';
export const DELETE_CARD = 'DELETE_CARD';

export const createCard = card => ({
  type: CREATE_CARD,
  card,
})

export const updateCard = card => ({
  type: UPDATE_CARD,
  card,
})

export const deleteCard = card => ({
  type: DELETE_CARD,
  cardId: card.id
})