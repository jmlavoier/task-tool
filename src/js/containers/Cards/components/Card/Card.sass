.card
  border-radius: 5px
  background-color: #95a372
  min-height: 100px
  padding: 10px
  maring-bottom: 5px
  margin-top: 5px