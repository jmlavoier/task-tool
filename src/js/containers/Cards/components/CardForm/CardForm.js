import React from 'react';
import PropTypes from 'prop-types';

import style from './CardForm.sass';

class CardForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: this.props.description,
    }
  }

  onDescriptionChange(event) {
    this.setState({
      description: event.target.value,
    })
  }

  onDescriptionKeyDown(event) {
    if (event.keyCode === 13) {
      this.props.onSaveCard({
        id: this.props.id,
        description: this.state.description,
      });
    }
  }

  onDescriptionBlur() {
    this.props.onSaveCard({
      id: this.props.id,
      description: this.state.description,
    });
  }

  componentDidMount() {
    if (this.el) {
      this.el.focus();
    }
  }

  componentDidUpdate() {
    if (this.el) {
      this.el.focus();
    }
  }

  render() {
    return (
      <textarea 
        ref={el => this.el = el}
        className={style['card-form']}
        value={this.state.description} 
        onChange={this.onDescriptionChange.bind(this)} 
        onKeyDown={this.onDescriptionKeyDown.bind(this)}
        onBlur={this.onDescriptionBlur.bind(this)}
      />
    );
  }
}

export default CardForm;