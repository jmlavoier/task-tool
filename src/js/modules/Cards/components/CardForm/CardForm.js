import React from 'react';
import PropTypes from 'prop-types';

import style from './CardForm.sass';

class CardForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      description: this.props.description,
    };

    this.onDescriptionChange = this.onDescriptionChange.bind(this);
    this.onDescriptionKeyDown = this.onDescriptionKeyDown.bind(this);
    this.onDescriptionBlur = this.onDescriptionBlur.bind(this);
  }

  componentDidMount() {
    if (this.el) {
      this.el.focus();
    }
  }

  componentDidUpdate() {
    if (this.el) {
      this.el.focus();
    }
  }

  onDescriptionChange(event) {
    this.setState({
      description: event.target.value,
    });
  }

  onDescriptionKeyDown(event) {
    // if (event.keyCode === 13) {
    //   this.props.onSaveCard({
    //     id: this.props.id,
    //     description: this.state.description,
    //   });
    // }
  }

  onDescriptionBlur() {
    this.props.onSaveCard({
      id: this.props.id,
      description: this.state.description,
    });
  }

  render() {
    return (
      <textarea
        ref={(el) => { this.el = el; }}
        className={style['card-form']}
        value={this.state.description}
        onChange={this.onDescriptionChange}
        onKeyDown={this.onDescriptionKeyDown}
        onBlur={this.onDescriptionBlur}
      />
    );
  }
}

CardForm.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  onSaveCard: PropTypes.func.isRequired,
};

export default CardForm;
