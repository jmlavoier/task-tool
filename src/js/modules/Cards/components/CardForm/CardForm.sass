.card-form
  outline: none
  border: 0
  background-color: rgba(0,0,0,0.2)
  width: calc(100% - 5px)
  height: 100px
  resize: none
  color: #f1f1f1
  font-size: 14px
  border-radius: 5px