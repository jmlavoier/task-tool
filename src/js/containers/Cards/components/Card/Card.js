import React from 'react';
import PropTypes from 'prop-types';
import { DragSource } from 'react-dnd';
import classNames from 'classnames';

import CardForm from 'containers/Cards/components/CardForm';
import style from './Card.sass';

/**
 * Implements the drag source contract.
 */
const cardSource = {
  beginDrag(props) {
    console.log(props);
    return {
      id: props.id, 
    };
  },

  endDrag(props, monitor, component) {
    if (!monitor.didDrop()) {
      return;
    }

    console.log(props, monitor, component);
  }
};

/**
 * Specifies the props to inject into your component.
 */
function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging()
  };
}

const getClassNames = (isDragging) => classNames({
  [style.card]: true,
  [style['is-dragging']]: isDragging,
});

const Card = ({ id, description, editMode, onSaveCard, onEditCard, isDragging, connectDragSource }) => {
  return connectDragSource(
    <div className={getClassNames(isDragging)} onClick={() => onEditCard(id)}  >
      {editMode 
        ? <CardForm id={id} description={description} onSaveCard={onSaveCard} />
        : description}
    </div>
  );
}

Card.propTypes = {
  description: PropTypes.string.isRequired,
  isDragging: PropTypes.bool.isRequired,
  connectDragSource: PropTypes.func.isRequired,
};

export default DragSource('CARD', cardSource, collect)(Card);
