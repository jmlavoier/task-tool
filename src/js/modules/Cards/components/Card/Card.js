import React from 'react';
import PropTypes from 'prop-types';
import { DragSource } from 'react-dnd';
import classNames from 'classnames';

import CardForm from 'modules/Cards/components/CardForm';
import style from './Card.sass';

const cardSource = {

  beginDrag(props) {
    /* istanbul ignore next */
    return {
      id: props.id,
    };
  },

  endDrag(props, monitor) {
    /* istanbul ignore next */
    if (!monitor.getDropResult()) {
      return;
    }
    /* istanbul ignore next */
    const { boardId } = monitor.getDropResult();

    /* istanbul ignore next */
    props.changeBoard(props.id, boardId);
  },
};

function collect(connect, monitor) {
  return {
    connectDragSource: connect.dragSource(),
    isDragging: monitor.isDragging(),
  };
}

const getClassNames = isDragging => classNames({
  [style.card]: true,
  [style['is-dragging']]: isDragging,
});

const Card = ({ id, description, editMode, onSaveCard, onEditCard, isDragging, connectDragSource }) => connectDragSource(
  <div role="presentation" className={getClassNames(isDragging)} onClick={() => onEditCard(id)} >
    {editMode
        ? <CardForm id={id} description={description} onSaveCard={onSaveCard} />
        : description}
  </div>,
);

Card.propTypes = {
  id: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  editMode: PropTypes.bool,
  onSaveCard: PropTypes.func.isRequired,
  onEditCard: PropTypes.func.isRequired,
  isDragging: PropTypes.bool.isRequired,
  connectDragSource: PropTypes.func.isRequired,
};

Card.defaultProps = {
  editMode: false,
};

export default DragSource('CARD', cardSource, collect)(Card);
