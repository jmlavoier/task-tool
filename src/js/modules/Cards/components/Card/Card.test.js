import React from 'react';
import uid from 'uid';
import { shallow, mount } from 'enzyme';
import TestBackend from 'react-dnd-test-backend';
import { DragDropContext } from 'react-dnd';
import CardForm from 'modules/Cards/components/CardForm';

import Card from './Card';

function wrapInTestContext(DecoratedComponent) {
  return DragDropContext(TestBackend)(props => <DecoratedComponent {...props} />);
}

test('Should component render with default props', () => {
  const OriginalBox = wrapInTestContext(Card);
  const identity = el => el;
  const description = 'Testing anything';

  const wrapper = mount(<OriginalBox connectDragSource={identity} id={uid()} onSaveCard={() => {}} onEditCard={() => {}} description={description} />);
  
  expect(wrapper.text()).toBe(description);
});


test('Should component render with props', () => {
  const OriginalBox = wrapInTestContext(Card);
  const identity = el => el;
  const description = 'Go to somewhere';
  
  const wrapper = mount(<OriginalBox connectDragSource={identity} id={uid()} onSaveCard={() => {}} onEditCard={() => {}} description={description} />);

  expect(wrapper.text()).toBe(description);
});


test('Should component render CardForm', () => {
  const OriginalBox = wrapInTestContext(Card);
  const identity = el => el;
  
  const wrapper = mount(<OriginalBox connectDragSource={identity} id={uid()} editMode onSaveCard={() => {}} onEditCard={() => {}} description="" />);

  expect(wrapper.find(CardForm)).toHaveLength(1);
})