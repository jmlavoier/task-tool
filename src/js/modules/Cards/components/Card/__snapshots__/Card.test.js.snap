// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component render description 1`] = `
<DragDropContext(Component)
  connectDragSource={[Function]}
  description="Testing anything"
  editMode={false}
  id="123"
  onEditCard={[Function]}
  onSaveCard={[Function]}
>
  <Component
    connectDragSource={[Function]}
    description="Testing anything"
    editMode={false}
    id="123"
    onEditCard={[Function]}
    onSaveCard={[Function]}
  >
    <DragSource(Card)
      connectDragSource={[Function]}
      description="Testing anything"
      editMode={false}
      id="123"
      onEditCard={[Function]}
      onSaveCard={[Function]}
    >
      <Card
        connectDragSource={[Function]}
        description="Testing anything"
        editMode={false}
        id="123"
        isDragging={false}
        onEditCard={[Function]}
        onSaveCard={[Function]}
      >
        <div
          className="card"
          onClick={[Function]}
          role="presentation"
        >
          Testing anything
        </div>
      </Card>
    </DragSource(Card)>
  </Component>
</DragDropContext(Component)>
`;
